File: leetcode/JavaScript实现/0082.js
// 自己的代码 ；第一想法就是非递归 一次循环搞定
var deleteDuplicates = function(head) {
    if(!head || !head.next)
        return head;
    let dummp = new ListNode(null);
    dummp.next = head;
    let pre = dummp, p = pre.next, chongfu = false;
    while(p.next) {
        if(p.val === p.next.val) {
            p = p.next;
            chongfu = true;
            if(!p.next) {
                pre.next = p.next;
                break;
            }
        } else if(chongfu) {
            pre.next = p.next;
            p = pre.next;
            chongfu = false;
        } else {
            pre = pre.next;
            p = pre.next;
        }
    }
    return dummp.next;
};

File: leetcode/C++实现/1-99题/0082.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head)
            return NULL;
        ListNode *temp=new ListNode(0);
        temp->next=head;
        ListNode *pre=temp,*p=pre->next;
        while(p->next)
        {
            if(p->next->val!=p->val)
            {
                if(pre->next==p)
                    pre=p;
                else
                    pre->next=p->next;
            }
            p=p->next;
        }
        if(pre->next!=p)
            pre->next=NULL;
        return temp->next;
    }
};
